expr/field_reference.go

```diff
@@ -540,6 +540,15 @@ func NewRootFieldRef(ref Reference, baseSchema *types.StructType) (*FieldReferen
 	return NewFieldRef(RootReference, ref, baseSchema)
 }
 
+// NewRootFieldRefFromType creates a new field reference with a specific known type.  Prefer using NewRootFieldRef.
+func NewRootFieldRefFromType(ref Reference, t types.Type) (*FieldReference, error) {
```

Comment:
Would it make sense to also have a `NewFieldRefFromType` function and just have this delegate to that (in the same way that `NewRootFieldRef` delegates to `NewFieldRef`)?

Reply:
Added.  It feels gutted if it isn't going to do any work like determine the type of an expression reference.